app/views/play/cinematic/Command/CinematicParser.js

```diff
@@ -13,33 +45,57 @@ export const parseShot = (shot, systems) => {
  * Returns an array of Commands for the setup of the shot.
  * @param {Shot} shot
  * @param {Systems} systems
+ * @returns {AbstractCommand[]} The commands to run at the start and end of the shot.
+ */
+const parseSetup = (shot, systems) =>
+  Object.values(systems)
+    .filter(sys => sys !== undefined && typeof sys.parseSetupShot === 'function')
+    .reduce((commands, sys) => {
+      const systemCommands = sys.parseSetupShot(shot)
+      if (!Array.isArray(systemCommands)) {
+        throw new Error('Your system should always return an array of commands.')
+      }
+      return [...commands, ...systemCommands]
+    }
+    , [])
+
+/**
+ * Returns an array of commands.
+ * @param {*} dialogNode 
+ * @param {*} systems 
  */
-const parseSetup = (shot, systems) => {
-  const { cinematicLankBoss, loader } = systems
-  const setupCommands = []
+const parseDialogNode = (dialogNode, systems) =>
+  Object.values(systems)
+    .filter(sys => sys !== undefined && typeof sys.parseDialogNode === 'function')
+    .reduce((commands, sys) => (
+      [...commands, ...sys.parseDialogNode(dialogNode)]
+    ), [])
 
-  const leftCharSlug = getLeftCharacterThangTypeSlug(shot)
+/**
+ * Parses a shot and dialog nodes.
+ *
+ * Returns a 2d array of command nodes. Each inner array is passed into the
+ * CommandRunner and run until user interuption or conclusion. Then between the
+ * inner arrays the cinematicController waits before running the next command batch.
+ *
+ * @param {Shot} shot The shot that is being run.
+ * @param {Object} systems The systems.
+ * @returns {AbstractCommand[][]} A 2d array of commands.
+ */
+export const parseShot = (shot, systems) => {
+  const setupCommands = parseSetup(shot, systems) || []
```

Comment:
Instead of sending all the systems to `parseSetup` and `parseDialogNode`, and then filtering there, I think it would be cleaner to filter here and then send only the specific system to `parseSetup` and `parseDialogNode`

Reply:
Can also probably rename `systems` to `cinematicSystems`?

Reply:
For now I will not change  the filtering (as I think I might change this a little more in the next PR). But I did make  the name more clear. Thanks!